src/sentry/rules/processing/processor.py

```diff
@@ -286,6 +317,9 @@ def apply_rule(self, rule: Rule, status: GroupRuleStatus) -> None:
             predicate_func = get_match_function(match)
             if predicate_func:
                 if not predicate_func(predicate_iter):
+                    # prevent return here if slow_conditions exist, because we have more evaluation to do
+                    if slow_conditions:
+                        continue
```

Comment:
```suggestion
                # prevent return here if slow_conditions exist, because we have more evaluation to do
                if not predicate_func(predicate_iter) and not slow conditions:
```

Reply:
nm I was reading it wrong